### PR TITLE
Code cleanup and optimization

### PR DESCRIPTION
**Pull request by jkmcl**
- Make 2 public static fields final
- Remove 1 unused private static final field
- Replace Stack with LinkedList

### CODE REVIEW

File: httpcore5/src/main/java/org/apache/hc/core5/net/URIBuilder.java [hunk 1]
```diff
@@ -1033,22 +1033,22 @@ public URIBuilder normalizeSyntax() {
         if (this.pathSegments != null) {
             final List<String> inputSegments = this.pathSegments;
             if (!inputSegments.isEmpty()) {
-                final Stack<String> outputSegments = new Stack<>();
+                final LinkedList<String> outputSegments = new LinkedList<>();
```
Comment (michael-o) [inline httpcore5/src/main/java/org/apache/hc/core5/net/URIBuilder.java:1036]: What is the benefit of this change?

Comment (jkmcl) [reply httpcore5/src/main/java/org/apache/hc/core5/net/URIBuilder.java:1036]: Performance. Stack extends Vector which is synchronized but LinkedList is not synchronized.

Comment (michael-o) [reply httpcore5/src/main/java/org/apache/hc/core5/net/URIBuilder.java:1036]: Right, yes. This makes then sense.

File: httpcore5/src/main/java/org/apache/hc/core5/util/Deadline.java [hunk 2]
```diff
@@ -60,12 +60,12 @@ public class Deadline {
...
      * The maximum (longest-lived) deadline.
      */
-    public static Deadline MAX_VALUE = new Deadline(INTERNAL_MAX_VALUE);
+    public static final Deadline MAX_VALUE = new Deadline(INTERNAL_MAX_VALUE);
 
     /**
      * The minimum (shortest-lived) deadline.
      */
-    public static Deadline MIN_VALUE = new Deadline(INTERNAL_MIN_VALUE);
+    public static final Deadline MIN_VALUE = new Deadline(INTERNAL_MIN_VALUE);
```
Comment (ok2c) [inline httpcore5/src/main/java/org/apache/hc/core5/util/Deadline.java:68]: @jkmcl This breaks binary compatibility. See build failure in Travis C. `URIBuilder` optimization looks good, but two first commits look useless. Please drop them.

**Review by ok2c — CHANGES_REQUESTED** [review]

File: httpcore5/src/main/java/org/apache/hc/core5/util/Deadline.java (hunk 2 above)
Comment (jkmcl) [reply httpcore5/src/main/java/org/apache/hc/core5/util/Deadline.java]: Sure, I've reverted those 2 commits. Please consider making `Deadline.MIN_VALUE` and `Deadline.MAX_VALUE` final in the future when the API is revised. Thanks

**Review by michael-o — APPROVED** [review]

**Review by ok2c — APPROVED** [review]

**Pull request merged**